code/game/objects/items/gems.dm

```diff
@@ -0,0 +1,134 @@
+//rare and valulable gems- designed to eventually be used for archeology, or to be given as opposed to money as loot. Auctioned off at export, or kept as a trophy. -MemedHams
+
+/obj/item/gem
+	name = "\improper Gem"
+	icon = 'icons/obj/gems.dmi'
+	icon_state = "rupee"
+
+	///Have we been analysed with a mining scanner?
+	var/analysed = FALSE
+	///How many points we grant to whoever discovers us
+	var/point_value = 100
+	///what's our real name that will show upon discovery? null to do nothing
+	var/true_name
+	///the message given when you discover this geyser.
+	var/analysed_message = null
+	///the thing that spawns in the item.
+	var/sheet_type = null
+
+/obj/item/gem/attackby(obj/item/item, mob/living/user, params) //Stolen directly from geysers, removed the internal gps
+	if(!istype(item, /obj/item/mining_scanner) && !istype(item, /obj/item/t_scanner/adv_mining_scanner))
+		return ..()
+
+	if(analysed)
+		to_chat(user, span_warning("This gem has already been analysed!"))
+		return
+
+	to_chat(user, span_notice("You analyse the precious gemstone!"))
+	if(analysed_message)
+		to_chat(user, analysed_message)
+
+	analysed = TRUE
+	if(true_name)
+		name = true_name
+
+	if(isliving(user))
+		var/mob/living/living = user
+
+		var/obj/item/card/id/card = living.get_idcard()
+		if(card)
+			to_chat(user, span_notice("[point_value] mining points have been paid out!"))
+			card.mining_points += point_value
+			playsound(src, 'sound/machines/ping.ogg', 15, TRUE)
+
+/obj/item/gem/welder_act(mob/living/user, obj/item/I) //Jank code that detects if the gem in question has a sheet_type and spawns the items specifed in it
+	if(I.use_tool(src, user, 0, volume=50))
+		if(src.sheet_type)
+			new src.sheet_type(user.loc)
+			to_chat(user, span_notice("You carefully cut [src]."))
+			qdel(src)
+		else
+			to_chat(user, span_notice("You can't seem to cut [src]."))
+	return TRUE
+
+/obj/item/gem/rupee
+	name = "\improper Ruperium Crystal"
+	desc = "A radioactive, crystalline compound rarely found in the goldgrubs. While able to be cut into sheets of uranium, the gem's true value is in its resonating, humming properties, often sought out by ethereal musicians to work into their gem-encrusted instruments. As a result, they fetch a fine price in most exchanges."
+	icon = 'icons/obj/gems.dmi'
+	icon_state = "rupee"
+	materials = list(/datum/material/uranium=20000)
+	sheet_type = /obj/item/stack/sheet/mineral/uranium{amount = 10}
+	point_value = 300
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/gem/fdiamond
+	name = "\improper Frost Diamond"
+	desc = "A unique diamond that is produced within icewing watchers. Rarely used in traditional marriage bands, various gemstone companies now try to effect a monopoly on it, to little success. It looks like it can be cut into smaller sheets of diamond ore that are normally found within watchers."
```

Comment:
```suggestion
	desc = "A unique diamond that is produced within icewing watchers. Rarely used in traditional marriage bands, various gemstone companies now try to effect a monopoly on it, to little success. It looks like it can be cut into smaller sheets of diamond ore."
```